### PR TITLE
enable ebpf dataplane for calico when specified

### PR DESCRIPTION
**Pull request by DockToFuture**
**How to categorize this PR?**

/area networking
/kind enhancement.

**What this PR does / why we need it**:
Calico can now be run in ebpf dataplane mode.
To enable it set `spec.networking.providerConfig.ebpfDataplane.enabled: true`.
It is recommended to  run in a pure networking mode with `spec.networking.providerConfig.ipv4.mode: Never` when using the ebpf dataplane.
The example file and docs are updated and it is forbidden to disable kube-proxy if not running in ebpf dataplane.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
Calico can now be run in ebpf dataplane mode. To enable it set `spec.networking.providerConfig.ebpfDataplane.enabled: true`. It is recommended to run in a pure networking mode with `spec.networking.providerConfig.ipv4.mode: Never` when using the ebpf dataplane. 
```

### CODE REVIEW

File: pkg/charts/charts_test.go [hunk 1]
```diff
@@ -174,6 +174,12 @@ var _ = Describe("Chart package test", func() {
 						"ipinip": map[string]interface{}{
 							"enabled": true,
 						},
+						"bpf": map[string]interface{}{
+							"enabled": false,
```
Comment (ScheererJ) [inline pkg/charts/charts_test.go:178]: Do you want to add any test with the bpf dataplane and/or the iptables cleanup being enabled?

File: pkg/controller/actuator_reconcile.go [hunk 2]
```diff
@@ -113,17 +113,21 @@ func (a *actuator) Reconcile(ctx context.Context, network *extensionsv1alpha1.Ne
...
 
-	if cluster.Shoot.Spec.Kubernetes.KubeProxy != nil && cluster.Shoot.Spec.Kubernetes.KubeProxy.Enabled != nil && !*cluster.Shoot.Spec.Kubernetes.KubeProxy.Enabled {
-		return field.Forbidden(field.NewPath("spec", "kubernetes", "kubeProxy", "enabled"), "Disabling kube-proxy is forbidden in conjunction with calico")
+	if cluster.Shoot.Spec.Kubernetes.KubeProxy != nil && cluster.Shoot.Spec.Kubernetes.KubeProxy.Enabled != nil && !*cluster.Shoot.Spec.Kubernetes.KubeProxy.Enabled && networkConfig.EbpfDataplane != nil && !networkConfig.EbpfDataplane.Enabled {
+		return field.Forbidden(field.NewPath("spec", "kubernetes", "kubeProxy", "enabled"), "Disabling kube-proxy is forbidden in conjunction with calico without running in ebpf dataplane")
 	}
 
+	kubeProxyEnabled := true
+	if cluster.Shoot.Spec.Kubernetes.KubeProxy != nil && cluster.Shoot.Spec.Kubernetes.KubeProxy.Enabled != nil && !*cluster.Shoot.Spec.Kubernetes.KubeProxy.Enabled {
+		kubeProxyEnabled = false
```
Comment (ScheererJ) [inline pkg/controller/actuator_reconcile.go:122]: How about removing the last check of the if clause and simply using it in the assignment?

```suggestion
		kubeProxyEnabled = *cluster.Shoot.Spec.Kubernetes.KubeProxy.Enabled
```

**Review by ScheererJ — CHANGES_REQUESTED** [review]
Looks good overall, just two minor changes.

File: pkg/charts/charts_test.go (hunk 1 above)
Comment (DockToFuture) [reply pkg/charts/charts_test.go:194]: Good point, added a test for the ebpf dataplane.

File: pkg/controller/actuator_reconcile.go (hunk 2 above)
Comment (DockToFuture) [reply pkg/controller/actuator_reconcile.go]: Adapted it. :)

**Review by rfranzke — APPROVED** [review]
/lgtm code-wise

**Pull request merged**